zerver/forms.py

```diff
@@ -155,6 +167,13 @@ def clean_realm_subdomain(self) -> str:
         check_subdomain_available(subdomain)
         return subdomain
 
+    def clean_realm_type(self) -> int:
+        realm_type = self.cleaned_data["realm_type"]
+        if realm_type not in self.REALM_TYPE_CHOICES:
+            return Realm.CORPORATE
```

Comment:
Is this the right error handling for an invalid type?  I think we could reasonably raise a ValidationError.

I suspect if we use https://docs.djangoproject.com/en/3.2/ref/models/fields/#enumeration-types (see the `models.IntegerField(choices=Suit.choices)` example) that would take care of this validation for us.